s3watcher/lib/Upload.js

```diff
@@ -44,16 +54,12 @@ module.exports = {
         const uploadRequest = request.post(options);
 
         return Rx.Observable.create(function(observer){
+            // check mime type, if not jpeg notify CW, stop!
+
             uploadRequest.on("response", function(response){
-                if(response.statusCode == 202){
-                    observer.onNext(response);
-                } else {
-                    const failMessage =
-                        "Failed upload with code: " + response.statusCode;
-                    observer.onError(failMessage);
-                }
-           });
-           uploadRequest.on("error", observer.onError.bind(observer));
+                observer.onNext(uploadResult(response));
+            });
+            uploadRequest.on("error", observer.onError.bind(observer));
```

Comment:
The result of this is that while timeouts will cause a Lambda failure and consequently retries. Any response from the server that is not `202` will log an error to Cloudwatch but not cause a Lambda failure, so no retry.